src/main/java/org/opentripplanner/routing/algorithm/mapping/RoutingResponseMapper.java

```diff
@@ -37,17 +37,24 @@ public static RoutingResponse map(
     // Create response
     var tripPlan = TripPlanMapper.mapTripPlan(request, itineraries);
 
-    var factory = mapIntoPageCursorFactory(
-      request.getItinerariesSortOrder(),
-      transitSearchTimeZero,
-      searchParams,
-      searchWindowForNextSearch,
-      firstRemovedItinerary,
-      request.pageCursor == null ? null : request.pageCursor.type
-    );
+    PageCursor nextPageCursor = null;
+    PageCursor prevPageCursor = null;
+
+    try {
+      var factory = mapIntoPageCursorFactory(
+        request.getItinerariesSortOrder(),
+        transitSearchTimeZero,
+        searchParams,
+        searchWindowForNextSearch,
+        firstRemovedItinerary,
+        request.pageCursor == null ? null : request.pageCursor.type
+      );
 
-    PageCursor nextPageCursor = factory.nextPageCursor();
-    PageCursor prevPageCursor = factory.previousPageCursor();
+      nextPageCursor = factory.nextPageCursor();
+      prevPageCursor = factory.previousPageCursor();
+    } catch (Exception e) {
```

Comment:
```suggestion
    } catch (IllegalArgumentException e) {
```

Reply:
A bit unsure if this is the right way to do this, this seems like the common case, so maybe it is better to move tha logging inside the mapIntoPageCursorFactory and return an NoopFactory (return `null` for next/previousPC).

Reply:
Changed the approach to not crash `mapIntoPageCursorFactory` when no path is found in heuristic search